### PR TITLE
Adds credit cost to printing items from the autolathe and protolathes

### PR DESCRIPTION
**Pull request by mudzbe**
## About The Pull Request

Adds a credit cost to printing items from the autolathe and protolathe. Autolathes bill to your ID/budget if you have it. Silicons using the autolathe bill to the civilian budget.
Protolathes bill to the department the lathe is from. Like the medlathe to the medbudget. The omnilathe doesn’t bill. New lathe boards have to be synced to a budget or they will enter the same pricing system as an autolathe, in which the printer pays instead of a budget.

## Why It's Good For The Game
Adds more value to credits and by extension budget cards.
Breaking into place is very casually and lightly punished by lack of enforcement. If the department directly suffers from unauthorised use of the protolathe.
Getting gamer gear becomes a conscious decision instead of “why not?” you will either have to budget yourself, charge people and do your job, steal it, or not buy it at all.
Buying a briefcase full of cash from the syndie uplink is no longer solely a meme.

## Changelog
:cl:
add: Added a credit cost to building items in the autolathe
add: Added a credit cost to building items in the protolathe

### CODE REVIEW

**Comment by mudzbe** [comment]
Some notes: 
- The changes in material_container.dm currently may or may not be unnecessary, I will look at it later, I thought I removed it.
- I am probably going to move the cost calculation to material_container.dm so it will be easy to reuse that code in the protolathe and also to cut down the repetition in autolathe.dm
- Variable material cost was very close to being added already but something in calling the list of material values broke, so I cut it. It was only ~2 lines of code so it would be easy to reprogram in once I figure out what went wrong

**Comment by mudzbe** [comment]
Also I need to figure out the correct costs everything has. Currently a hatchet and .36 speed loader is 400, shotgun shells are around ~30 a shell and general things range from 10-40 credits. I don’t think people will be printing many items, so this price range with the average salary seems to work out. I could reduce the cost of the speedloader or add more on round start to the set locker, then again the det shouldn’t be shooting everyone they see.

**Comment by jbox144** [comment]
Ehhh... not the biggest fan. Of this. What about ghost roles, borgs, drones, e.t.c which also use them? I'd be fine if you could set up protolathes and autolathes with an "ownership", which means you pay to that account to print an item (for example, using the cargo autolathe pays to the cargo budget, or you can adjust the price to not cost anything if you want). It'd still instigate some financial flow, without forcing it.

**Comment by mudzbe** [comment]
> Ehhh... not the biggest fan. Of this. What about ghost roles, borgs, drones, e.t.c which also use them? I'd be fine if you could set up protolathes and autolathes with an "ownership", which means you pay to that account to print an item (for example, using the cargo autolathe pays to the cargo budget, or you can adjust the price to not cost anything if you want). It'd still instigate some financial flow, without forcing it.

I previously had an else statement to the ishuman() check where it would charge the bill to the civilian budget, I removed it somewhere along the way, but I’ll readd it. Protolathes will charge to their respective department, which should be easy. As for charging to specific departments for autolathes I will look into it tomorrow arvo but that might include having to do map changes, unless there’s already a variable for that there already. I’m pretty sure there’s a system already for creating new bank accounts, I’ve seen something along those lines in the code. The aim is for people to charge for gamer gear and to make the consequences for breaking into places more severe instead of how it is which is very casual. Stealing tech disks from science will be actually a challenge as people are probably going to block off the entrance from maint into RND.

**Comment by mudzbe** [comment]
I will fill out the about the PR and why it’s good for the game tab later

**Comment by TheFakeElon** [comment]
Hmm, of this is modularized and austation notes are lackluster. I'd recommend either finding a way to modularize it or PRing upstream instead.

File: code/datums/components/material_container.dm [hunk 1]
```diff
@@ -280,19 +280,20 @@
 		return FALSE
 
 	for(var/x in mats) //Loop through all required materials
-		var/datum/material/req_mat = x
-		if(!istype(req_mat))
-			if(ispath(req_mat)) //Is this an actual material, or is it a category?
-				req_mat = getmaterialref(req_mat) //Get the ref
+		if(!x == "Credits")
```
Comment (TheFakeElon) [inline code/datums/components/material_container.dm:283]:
```suggestion
		if(x != "Credits")
```

Comment (mudzbe) [reply code/datums/components/material_container.dm:283]: That was there before the PR kinda rude why do I gotta fix it

**Comment by mudzbe** [comment]
> Ehhh... not the biggest fan. Of this. What about ghost roles, borgs, drones, e.t.c which also use them? 
Wait a second, I already have support for silicons you liar

**Comment by mudzbe** [comment]
Gonna port this shit to bee

**Comment by mudzbe** [comment]
gonna reopen if I care again

**Pull request closed without merging**